src/js/pages/ArbitrationLicense/index.jsx

```diff
@@ -0,0 +1,106 @@
+import React, {Component} from 'react';
+import {withRouter} from "react-router-dom";
+import PropTypes from 'prop-types';
+import {connect} from "react-redux";
+
+import arbitration from "../../features/arbitration";
+import network from "../../features/network";
+import metadata from "../../features/metadata";
+
+import Info from './components/Info';
+import BuyButton from './components/BuyButton';
+import Balance from './components/Balance';
+import Loading from '../../components/Loading';
+import ErrorInformation from '../../components/ErrorInformation';
+
+const LICENSE_TOKEN_SYMBOL = 'SNT';
+
+class ArbitrationLicense extends Component {
+  componentDidMount() {
+    if (this.props.isLicenseOwner) {
+      this.props.loadProfile(this.props.address);
+      return this.props.history.push('/');
+    }
+
+    this.props.checkLicenseOwner();
+    this.props.loadLicensePrice();
+    this.props.updateBalance(LICENSE_TOKEN_SYMBOL);
+  }
+
+  componentDidUpdate() {
+    if (this.props.isLicenseOwner) {
+      this.props.loadProfile(this.props.address);
+      return this.props.history.push('/');
+    }
+  }
+
+  buyLicense = () => {
+    this.setState({isBuying: true});
+    this.props.buyLicense();
+  };
+
+  enoughBalance() {
+    return this.props.sntToken && parseInt(this.props.sntToken.balance, 10) >= parseInt(this.props.licensePrice, 10);
+  }
+
+  render() {
+    if (this.props.isLoading) {
+      return <Loading mining/>;
+    }
+    console.log(this.props);
+
+    if (!this.props.sntToken) {
+      return <ErrorInformation sntTokenError retry={this.buyLicense}/>;
+    }
+    if (this.props.isError) {
+      return <ErrorInformation transaction retry={this.buyLicense}/>;
+    }
```

Comment:
I would put the errors above the Loading, just in case we could end up with a forever loading